src/onnx/onnx.cpp

```diff
@@ -745,26 +745,72 @@ struct onnx_parser
         op::slice op;
         std::vector<size_t> dims = args[0]->get_shape().lens();
         size_t num_dims          = dims.size();
-        if(contains(attributes, "axes"))
+
+        // slice can have up to 5 inputs, we first check the 5th one
+        // to decide whether MIGRAPHX can handle this slice
+        if(args.size() == 5)
+        {
+            auto tind_ins = args.back();
+            if(!tind_ins->can_eval())
```

Comment:
Can we make a function that checks can_eval and replace the conditionals on this line and 770 and 786?

Reply:
we have a function check_arg_empty() that can be used here.